onnxruntime/core/providers/nnapi/nnapi_builtin/nnapi_execution_provider.cc

```diff
@@ -53,7 +54,11 @@ NnapiExecutionProvider::GetCapability(const onnxruntime::GraphViewer& graph_view
   }
 
   nnapi::ModelBuilder builder(graph_view);
-  const auto supported_nodes_vector = builder.GetSupportedNodes();
+  nnapi::OpSupportCheckParams params{
+      builder.GetAndroidSdkVer(),
```

Comment:
This is the only part I cannot get rid of ModelBuilder, how do we pass the target_android_sdk_ver to NNAPI EP?

Add the target_android_sdk_ver to NNAPI EP ctor
Use session config options
Compile def?
Will address this in next PR